tron/core/actionrun.py

```diff
@@ -450,6 +525,8 @@ def _done(self, target, exit_status=0):
             self.clear_trigger_timeout()
             self.exit_status = exit_status
             self.end_time = timeutils.current_time()
+            if self.last_attempt is not None and self.last_attempt.end_time is None:
```

Comment:
here and in other places: is it possible to have `.stop` called when there is no `last_attempt`? maybe worth just bailing at the start of the method with an error?

Reply:
Yeah, it's possible if someone uses `tronctl fail` on a run that hasn't started yet. I think it's safer to just handle it.